routes/ppe.js

```diff
@@ -64,39 +72,99 @@ router.post('/',
     check("uri", "hyperlink 'uri' must be present and conform to a URL").if((value, { req }) => { return req.body.kind == "hyperlink"; }).exists().isURL()
   ]),
   async function (req, res, next) {
-
     try {
+      let transporter = await require('../config/mailer')
       const proof = await models.Proof.create();
       const document = await models.Document.create({
         kind: req.body.kind,
         uri: makeURI(req.body.kind, req.body.uri, req.file),
         ProofId: proof.id
       });
-      let record = {
-        name: req.body.name,
-        PPETypeId: req.body.PPETypeId,
-        quantity: req.body.quantity,
-        email: req.body.email,
-        contact: req.body.contact,
-        latitude: req.body.latitude,
-        longitude: req.body.longitude,
-        ProofId: proof.id
+      const [user, created] = await models.User.findOrCreate({
+        where: { email: req.body.email },
+        defaults: {
+          username: req.body.name,
+          role: 'Applicant',
+          contact: req.body.contact,
+          password: await argon2.hash(req.body.name),
+          createdAt: new Date(),
+          updatedAt: new Date(),
+          token: cryptoRandomString({ length: 50, type: 'url-safe' })
+        }
+      });
+      // send mail with defined transport object
+      const statusLink = `${process.env.BASE_URL || 'http://localhost:3000/'}user/${user.token}`;
+      let info = transporter.sendMail({
+        from: process.env.EMAIL_SENDER || 'PPE Tracker', // sender address
+        to: user.email, // list of receivers
+        subject: "Your PPE Application", // Subject line
+        text: `Your application was registered successfully. Please open this link to view your application status: ${statusLink}`, // plain text body
+        html: `
+          <h1>Your application was registered successfully</h1>
+          <p>Please click on this link to view your application status</p>
+          <a href="${statusLink}" target="_blank">${statusLink}</a>
+          `
+      }).then(()=>{
+        console.log("Message sent: %s", info.messageId);
+        // Message sent: <b658f8ca-6296-ccf4-8306-87d57a0b4321@example.com>
+        // Preview only available when sending through an Ethereal account
+        console.log("Preview URL: %s", nodemailer.getTestMessageUrl(info));
+        // Preview URL: https://ethereal.email/message/WaQKMgKddxQDoou...  
+      });
+
+      console.log("USER: " + user.id)
```

Comment:
Unnecessary, IMO the app should save the token in a cookie. Perhaps, that's been planned?